storage/sectorstorage/roprov.go

```diff
@@ -0,0 +1,25 @@
+package sectorstorage
+
+import (
+	"context"
+
+	"github.com/filecoin-project/go-sectorbuilder"
+	"github.com/filecoin-project/lotus/storage/sectorstorage/stores"
+
+	"github.com/filecoin-project/specs-actors/actors/abi"
+	"golang.org/x/xerrors"
+)
+
+type readonlyProvider struct {
+	stor *stores.Local
+}
+
+func (l *readonlyProvider) AcquireSector(ctx context.Context, id abi.SectorID, existing sectorbuilder.SectorFileType, allocate sectorbuilder.SectorFileType, sealing bool) (sectorbuilder.SectorPaths, func(), error) {
+	if allocate != 0 {
```

Comment:
magic number logic here is hard to follow

Reply:
Yeah, constants would be great.